returnn/tf/util/basic.py

```diff
@@ -1339,6 +1339,33 @@ def __call__(self, shape, dtype=None, partition_info=None):
   return LoadTxtFileInitializer()
 
 
+class GammatoneFilterbankInitializer(init_ops.Initializer):
+  """
+  Initializer for a gammatone filterbank, e.g., to initialize weights of a convolutional layer.
+  """
+
+  def __init__(self, **kwargs):
+    """
+    :param kwargs: kwargs for GammatoneFilterbank
+    """
+    from returnn.util.sig_proc import GammatoneFilterbank
+    self.gammatone_filterbank = GammatoneFilterbank(**kwargs)
+
+  def __call__(self, shape, dtype=None, partition_info=None):
+    """
+    :param tuple[int] shape:
+    :param tf.DType|None dtype:
+    :param partition_info:
+    :rtype: tf.Tensor
+    """
+    import numpy
+    fbank = self.gammatone_filterbank.get_gammatone_filterbank()
+    fbank = numpy.reshape(fbank, shape)
```

Comment:
Normally, an initializer should be generic for any shape. But in your case, it looks like any other shape than (sample_rate * length, num_channels) this is not possible, right? Then this should rather be a check `assert shape == (...)` instead of doing a reshape here.

Reply:
Basically that's right. However, in my case e.g. the desired shape has another axis of dimension one which is then added by this reshape. I can still do an assertion but I'm not sure if `sample_rate * length` is always the first and `num_channels` is always the last axis.

Reply:
Note that if you reorder it, you would get very wrong results then, and would not even notice.
Adding a 1-dim axis somewhere is always fine, though.
Not sure sure if you want to get clever about the check, or maybe the reshape is fine.

Reply:
Yes, reordering would indeed be a problem. I'll add the assertion with first and last axis. Once this becomes an issue we can check it. Better this way then not noticing possible problems.